### DISCUSSION TITLE
Multi Maps: Artifacts due to unfavorable map draw priority

### DISCUSSION THREAD

**Discussion by Klaus-Tockloth** [mapsforge/mapsforge#1334] [General]
This map (DEU+NORD) has some rendering artifacts at the map border:
<img width="1011" alt="Bildschirmfoto 2022-07-09 um 14 12 43" src="https://user-images.githubusercontent.com/5493128/178105508-6866fdff-1b9b-42ec-b0dc-6533201ceb39.png">

This map (DEU+WEST) renders the object "Schloss Nordkirchen" correct:
<img width="1011" alt="Bildschirmfoto 2022-07-09 um 14 13 04" src="https://user-images.githubusercontent.com/5493128/178105554-6cd61960-4aff-425b-8655-da261338f1d9.png">

The combination of both maps (DEU+NORD and DEU+WEST) gives DEU+NORD drawing priority. This leads to incorrect rendering of the object "Schloss Nordkirchen":
<img width="1011" alt="Bildschirmfoto 2022-07-09 um 14 13 25" src="https://user-images.githubusercontent.com/5493128/178105631-102c00cf-87cc-4894-8e8f-07ddea12f411.png">

**Comment by devemux86** [comment]
That's how Mapsforge multiple maps work.
They cannot be perfect at the overlapping areas where multiple (and same) content is combined.

More details exist in [documentation](https://github.com/mapsforge/mapsforge/blob/master/docs/MapDataStore.md#multimapdatastore).
You can try the `DEDUPLICATE` option with Mapsforge samples.

Or better try [VTM](https://github.com/mapsforge/vtm) in [Cruiser GL](https://wiki.openstreetmap.org/wiki/Cruiser) to see if the rendering is better.

**Reply by Klaus-Tockloth** [reply]
> That's how Mapsforge multiple maps work.
> They cannot be perfect at the overlapping areas where multiple (and same) content is combined.

It needs some logic to decide which map to draw on top. Probably difficult.

<img width="1011" alt="Bildschirmfoto 2022-07-10 um 09 47 11" src="https://user-images.githubusercontent.com/5493128/178136142-3fb1e25d-36d4-4f4e-aa3f-88380c40bb0d.png">

versus

<img width="1076" alt="Bildschirmfoto 2022-07-10 um 09 48 27" src="https://user-images.githubusercontent.com/5493128/178136156-fccd8a73-9ff3-4eb3-8e03-cfb403d8e7dc.png">

**Comment by devemux86** [comment]
There is no z-order logic in map files.
The default [RETURN_ALL](https://github.com/mapsforge/mapsforge/blob/master/docs/MapDataStore.md#multimapdatastore) policy simply combines the data from all map tiles at the same region.

Z-order can work in map layers, but that is for apps (not library) and their custom implementations.

**Reply by Klaus-Tockloth** [reply]
> There is no z-order logic in map files.
> The default [RETURN_ALL](https://github.com/mapsforge/mapsforge/blob/master/docs/MapDataStore.md#multimapdatastore) policy simply combines the data from all map tiles at the same region.

OK, that blocks a solution. My conclusion: The "Multi Map Feature" results  under some rare conditions to confusing render artifacts.
